bundles/automation/org.eclipse.smarthome.automation.module.core/src/main/java/org/eclipse/smarthome/automation/module/core/handler/GenericEventTriggerHandler.java

```diff
@@ -87,6 +91,15 @@ public void receive(Event event) {
             if (!event.getTopic().contains(source)) {
                 return;
             }
+            // if the optional parameter "eventOnChannel" is set different from the defaultValue (""), compare it to the
+            // received event on the channel
+            if (eventOnChannel != null && !eventOnChannel.equals("")) {
+                if (event instanceof ChannelTriggeredEvent) {
```

Comment:
line 96 to 98 can be reduced to `if (StringUtils.isNotBlank(eventOnChannel) && event instanceof ChannelTriggeredEvent && !eventOnChannel.equals(((ChannelTriggeredEvent) event).getEvent())) {`

This may also be a private method `private boolean eventMatchesChannelConfiguration(event)` so it will only be one if clause around the early return.